packages/web3-eth/src/utils/format_transaction.ts

```diff
@@ -1,11 +1,16 @@
-import { convertToValidType, Numbers, ValidReturnTypes, ValidTypes, mergeDeep } from 'web3-utils';
+import { DataFormat, DEFAULT_RETURN_FORMAT, format, FormatType } from 'web3-common';
+import { bytesToBuffer, mergeDeep } from 'web3-utils';
 import { TransactionDataAndInputError } from '../errors';
+import { transactionSchema } from '../schemas';
 import { Transaction } from '../types';
 
 export function formatTransaction<
-	DesiredType extends ValidTypes,
-	NumberType extends ValidReturnTypes[DesiredType] = ValidReturnTypes[DesiredType],
->(transaction: Transaction, desiredType: DesiredType): Transaction<NumberType> {
+	ReturnFormat extends DataFormat = typeof DEFAULT_RETURN_FORMAT,
+	TransactionType extends Transaction = Transaction,
+>(
+	transaction: TransactionType,
+	returnFormat: ReturnFormat = DEFAULT_RETURN_FORMAT as ReturnFormat,
+): FormatType<TransactionType, ReturnFormat> {
 	const formattedTransaction = mergeDeep(
```

Comment:
```suggestion
	let formattedTransaction = mergeDeep(
```

Reply:
We didn't reassigned the `formattedTransaction`, so `const` is good to use.